### DISCUSSION TITLE
Can I pass data from my shell to a kitten?

### DISCUSSION THREAD

**Discussion by groves** [kovidgoyal/kitty#5202] [Q&A] [answered]
In a kitten I'm writing I'd like to know what command my fish shell is running regardless if it's a local or remote shell. I'm using the `kitty-kitten-result` escape elsewhere to pass data back from remote back to a kitten. That had me thinking I could use the [`fish_prexec` and `fish_postexec` events](https://fishshell.com/docs/current/cmds/function.html?highlight=preexec) to emit an escape code and the executing command that I then pick up in my kitten. I can't use `kitten-result` though as it'll get clobbered by other things. Is there another way to get data back and accessible to a kitten?

**Comment by kovidgoyal** [comment]
Doesnt shell integration already do this? just check th etitle of the
window the shell is running in, it is set to either cwd of the shell or
a command, should be easy enough to distnguish between the two, using
last_reported_cwd

On Thu, Jun 16, 2022 at 01:57:00PM -0700, Charlie Groves wrote:
> In a kitten I'm writing I'd like to know what command my fish shell is running regardless if it's a local or remote shell. I'm using the `kitty-kitten-result` escape elsewhere to pass data back from remote back to a kitten. That had me thinking I could use the [`fish_prexec` and `fish_postexec` events](https://fishshell.com/docs/current/cmds/function.html?highlight=preexec) to emit an escape code and the executing command that I then pick up in my kitten. I can't use `kitten-result` though as it'll get clobbered by other things. Is there another way to get data back and accessible to a kitten?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/5202
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by groves** [reply]
The title also contains stuff like the hostname. I could write something to parse that, but it feels fragile. I'd rather spit out the exact data I want in a place specific to my use of it.

**Comment by kovidgoyal** [comment]
You can configure fish to output whatever you like as the title. 
Should be easy to use a format that both looks good to a human and is
robustly parseable.

But if you dont want to do that simply monkey patch
Window.handle_kitten_result then you can use it more different kinds of
data.

**Comment by kovidgoyal** [comment, marked as answer]
And this avoids the need to monkeypatch: https://github.com/kovidgoyal/kitty/commit/bf01c549f5a7a2484b0983755cfd73b8cd94ab91

**Reply by groves** [reply]
This looks great! Can't wait to try it.

**Reply by kovidgoyal** [reply]
You shouldnt need to wait, its in the nightly.
